tools/reckless

```diff
@@ -0,0 +1,684 @@
+                                'bitcoin-reckless.conf')
+        if default_text is None:
+            default_text = '# This configuration file is managed by reckles' +\
+                           's to activate and disable\n# reckless-installed' +\
+                           ' plugins\n\n'
+        Config.__init__(self, path=str(path), default_text=default_text)
+        self.reckless_dir = os.path.split(path)[0]
+
+
+class LightningBitcoinConfig(Config):
+    """lightningd config specific to the bitcoin network. This is inherited by
+    the main lightningd config and in turn, inherits bitcoin-reckless.conf."""
+
+    def __init__(self, path=None, default_text=None, warn=True):
+        if path is None:
+            path = os.path.join(LIGHTNING_DIR, 'bitcoin', 'config')
+        if default_text is None:
+            default_text = "# This config was autopopulated by reckless\n\n"
+        Config.__init__(self, path=str(path),
+                        default_text=default_text, warn=warn)
+
+
+class InferInstall():
+    """Once a plugin is installed, we may need its directory and entrypoint"""
+    def __init__(self, name):
+        reck_contents = os.listdir(RECKLESS_CONFIG.reckless_dir)
+        if name[-3:] == '.py':
+            name = name[:-3]
+        if name in reck_contents:
+            self.dir = os.path.join(RECKLESS_CONFIG.reckless_dir, name)
+        else:
+            raise Exception(f"Could not find a reckless directory for {name}")
+        plug_contents = os.listdir(os.path.join(RECKLESS_CONFIG.reckless_dir,
+                                                name))
+        for n in py_entry_guesses(name):
+            if n in plug_contents:
+                self.entry = os.path.join(self.dir, n)
+                self.name = n
+                return
+        raise Exception(f'plugin entrypoint not found in {self.dir}')
+
+
+def help(target):
+    if len(target) > 0:
+        print(globals()[target[0]].__doc__)
+    else:
+        parser.print_help(sys.stdout)
+
+
+def verbose(*args):
+    if not IS_VERBOSE:
+        return
+    print(*args)
+
+
+def _search_repo(name, url):
+    """look in given repo and, if found, populate InstInfo"""
+    # Remove api subdomain, subdirectories, etc.
+    repo = url.split('/')
+    while '' in repo:
+        repo.remove('')
+    repo_name = None
+    for i in range(len(repo)):
+        if 'github.com' in repo[i]:
+            # Extract user and repo name
+            start = i + 1
+            if repo[start] == 'repo':
+                # Maybe we were passed an api.github.com/repo/<user> url
+                start = start + 1
+            repo_user = repo[start]
+            repo_name = repo[start+1]
+            break
+    # FIXME: Handle non-github repos.
+    # Get details from the github API.
+    if repo_name is not None:
+        api_url = f'https://api.github.com/repos/{repo_user}/' + \
+                  f'{repo_name}/contents/'
```

Comment:
Python can auto-concatenate string continuations, no need for `+ \`:

```suggestion
        api_url = (
            f'https://api.github.com/repos/{repo_user}/'
            f'{repo_name}/contents/'
        )
```

Gives back some of the characters that it is using on the whitespace indentation :-)